interview-questions.md

```diff
@@ -8,33 +8,64 @@ Then, research the question to expand on your answer. Even if you feel you have
+Your answer: Yarn is Yet Another Resource Negotiator. the files that are modified when runnin yarn are the node module files to add dependencies for related files and the yarn lock file that is created.
 
 Researched answer:
 
+(The Source: https://en.wikipedia.org/wiki/Yarn_(package_manager))
+
+(language from the source: Yarn is one of the main JavaScript package managers,developed in 2016 by Meta (formerly Facebook) for the Node.js JavaScript runtime environment. An alternative to the npm package manager, Yarn was created as a collaboration of Facebook, Exponent (now Expo.dev), Google, and Tilde (the company behind Ember.js) to solve consistency, security, and performance problems with large codebases.)
+
+(My Interpretation: Yarn as it's name states is Yet Another Resource negotiator. It is a package manager that will find, download and install files that you need for certain task. Looking into what yarn is ultimatily wasnt as frutifull in helping me understand as looking into what a package manager is. Because that is ultimatly what yarn is.)
+
 3. What is a React component?
 
-Your answer:
+Your answer: A react component is part if an overall react build. Best prectices seems to be to separate your compnentents for their specific purpose to prevent corrumption of the overall build.
 
-Researched answer:
+Researched answer: 
+(the source: https://www.w3schools.com/react/react_components.asp#:~:text=Components%20are%20independent%20and%20reusable,will%20concentrate%20on%20Function%20components.)(secondary source: Manager XD Development Team at Illumina)
+
+(language from the source: Components are independent and reusable bits of code. They serve the same purpose as JavaScript functions, but work in isolation and return HTML.
+Components come in two types, Class components and Function components, in this tutorial we will concentrate on Function components.)
+
+(My interpretation: Components are are part of the build that are segrigated to make the code more D.R.Y and are resubale and make the code easier to read)
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: State values are like a save point in a video game. At the start of the game the you spawn at the start of the level every time(current state). Till you hit a save point which changes your spawn point (setting a new state) for where you will "start the level" each time after that. Props are like the Portal Gun. They are arguments that are called between react components.
 
 Researched answer:
+(the source:https://www.w3schools.com/react/react_props.asp )
+
+(language from the source: Props are arguments passed into React components.
+The state object is where you store property values that belongs to the component.
+When the state object changes, the component re-renders.)
+
+(My Interpretation: Props allow you to store properties in a diffent component and call them to the main file build when needed. keepying your code D.R.Y
```

Comment:
State allows you to store properties in a different component, and props allow you to pass and reference those properties as needed.